src/scripts/renderPages.js

```diff
@@ -0,0 +1,172 @@
+const DEFAULT_FIRST_ELEMENT = 1;
+const MIN_ELEMENTS_TO_SHOW_FROM_LEFT = 4;
+const MIN_ELEMENTS_TO_SHOW_FROM_RIGHT = 3;
+const MIN_ELEMENTS_BEFORE_ACTIVE_ELEMENT = 2;
+const MIN_ELEMENTS_AFTER_ACTIVE_ELEMENT = 1;
+const VISIBLE_ELEMENTS_AFTER_FIRST_ELEMENT_ON_MIDDLE = 3;
+const MAX_ELEMENTS_TO_DISPLAY_ALL_ELEMENTS = 6;
+const MIN_VALUE_FROM_RIGHT_TO_SHOW_RIGHT_DOTS = 2;
+
+const getPaginationStartWhenLeftDots = (currentElement, maxElement) => {
+    return Math.min(
+        currentElement - MIN_ELEMENTS_BEFORE_ACTIVE_ELEMENT,
+        maxElement - MIN_ELEMENTS_TO_SHOW_FROM_RIGHT
+    );
+};
+
+const getPaginationStartWhenRightDots = currentElement => {
+    return Math.max(
+        currentElement + MIN_ELEMENTS_AFTER_ACTIVE_ELEMENT,
+        MIN_ELEMENTS_TO_SHOW_FROM_LEFT
+    );
+};
+
+const getPaginationStartWhenBothDots = currentElement => {
+    return currentElement - MIN_ELEMENTS_BEFORE_ACTIVE_ELEMENT;
+};
+
+const getPaginationEndWhenBothDots = paginationStart => {
+    return paginationStart + VISIBLE_ELEMENTS_AFTER_FIRST_ELEMENT_ON_MIDDLE;
+};
+
+const getFirstElements = () => {
+    const firstElements = /* HTML */ `
+        <li
+            class="pagination-component__page"
+            id="1"
+            data-testid="pagination-page"
+        >
+            <a class="pagination-component__link">1</a>
+        </li>
+        <li
+            class="
+            pagination-component__page"
+            data-testid="pagination-page"
+        >
+            <a class="pagination-component__link">...</a>
+        </li>
+    `;
+    return firstElements;
+};
+
+const getLastElements = maxElement => {
+    const lastElements = /* HTML */ `
+        <li class="pagination-component__page" data-testid="pagination-page">
+            <a class="pagination-component__link">...</a>
+        </li>
+        <li
+            data-testid="pagination-page"
+            class="
+            pagination-component__page"
+            id="${maxElement}"
+        >
+            <a class="pagination-component__link">${maxElement}</a>
+        </li>
+    `;
+    return lastElements;
+};
+
+const getPaginationElements = (
+    paginationStart,
+    paginationEnd,
+    currentElement
+) => {
+    let paginationPages = ``;
+
+    for (let counter = paginationStart; counter <= paginationEnd; counter++) {
```

Comment:
no ta pętla trochę tajemnicza dla mnie :D
lepiej jakby się to udało zapisać jakimś Array.forEach em